Source/JavaScriptCore/API/PASReportCrash.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <stdio.h>
+#include <JavaScriptCore/JSExportMacros.h>
+#include <bmalloc/pas_report_crash.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if BENABLE(LIBPAS) && OS(DARWIN)
+JS_EXPORT_PRIVATE kern_return_t PASReportCrashExtractResults(vm_address_t fault_address, mach_vm_address_t pas_dead_root, task_t task, pas_report_crash_pgm_report *report, crash_reporter_memory_reader_t crm_reader);
```

Comment:
You should use `JS_EXPORT` (like other API and SPI functions) instead of `JS_EXPORT_PRIVATE`.

The declaration need to have `JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA)) at the end of it.

There's tension here with this being libpas SPI and not JSC SPI.  The only reason this file is in the JSC API directory is because it is packaged as part of the JSC framework.  With that ...

It seems odd that we're working with a `pas_report_crash_pgm_report` struct follows libpas style while the function name follows JSC style.  How about we just call this function `pas_report_crash_extract_results` instead and go all in on being consistent with libpas style rather than being halfway there.

Also let's call this file `PasReportCrashPrivate.h` to indicate that it is SPI.  It should not be included in the public SDK.

Reply:
Also, this SPI should take a `size_t reportSize` argument after the `report` argument.  See `pas_report_crash_extract_pgm_failure` for more details.